server/database/users.go

```diff
@@ -35,3 +35,21 @@ func GetPLayerIdForUser(db *sql.DB, userId int64, player string) (int64, error)
 	}
 	return id, nil
 }
+
+func IsObjectOwnedByPlayer(db *sql.DB, playerId int64, objectId int64) (bool, error) {
+	ownedObjects := make(map[int64]struct{})
+	rows, err := db.Query("select `id` from t_object where owner = ?", playerId)
```

Comment:
this looks like you have turned the test upside down.

`select owner from t_object where id = ?`

and test the returned value for equivalence with playerId. no loops needed.

while removing this loop is improvement, see the other comment for actual solution.

Reply:
resolved in another comment